Database/Entities/PetEnitiy.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace Database.Entities
+{
+    public class PetEnitiy
+    {
+        [Key]
+        public required Guid Id { get; init; }
+
+        // foreign key 
+        public required ShelterEntity Shelter { get; init; } = null!;
```

Comment:
It's better to use `[Required]` attribute here instead. `required` keyword forces us to set this property during object initialization. We don't need to do that, because we can simply set `ShelterId` instead, and EF core will register the relation correctly (because, as you see in migration, there is no column corresponding to `Shelter` property in db).